aiven/client/client.py

```diff
@@ -248,7 +250,7 @@ def update_service_connection_pool(
     ):
         body = {}
         if username is not None:
-            body["username"] = username
+            body["username"] = None if username == "" else username
```

Comment:
It could be better to keep the "empty string means set to None" in `cli.py` and not the `AivenClient`.

This would mean we can't use `None` as the default value to detect "parameter unspecified" here, and would need a sentinel object:
```python
    UNCHANGED = object()
    def update_service_connection_pool(
        self,
        project,
        service,
        pool_name,
        dbname=None,
        username=UNCHANGED,
        pool_size=None,
        pool_mode=None,
    ):
        body = {}
        if username is not UNCHANGED:
            body["username"] = username
```
Overall this will be more verbose (the cli.py part will be less trivial), but making the `"" to None` conversion on the cli side will help us write tests similar to https://github.com/aiven/aiven-client/blob/master/tests/test_cli.py#L123 . Having logic inside `AivenClient` is harder to test.

Reply:
Yes, that's probably better. Changed and added some tests.